gdx/test/com/badlogic/gdx/utils/LongArrayTest.java

```diff
@@ -0,0 +1,425 @@
+		longArray2.addAll(new long[] {1, 3, 4, 5, 6});
+		longArray2.insert(1, 2);
+		Assert.assertArrayEquals(new long[] {1, 2, 4, 5, 6, 3}, longArray2.toArray());
+		try {
+			longArray2.insert(2783, 3);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to insert at an index out of bounds
+		}
+	}
+
+	/** Test of the swap() method */
+	@Test
+	public void swapTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6});
+		longArray1.swap(1, 4);
+		Assert.assertArrayEquals(new long[] {1, 6, 4, 5, 3}, longArray1.toArray());
+		try {
+			longArray1.swap(100, 3);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to swap at an index out of bounds
+		}
+		try {
+			longArray1.swap(3, 100);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to swap at an index out of bounds
+		}
+	}
+
+	@Test
+	public void containsTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6});
+		Assert.assertTrue(longArray1.contains(3));
+		Assert.assertFalse(longArray1.contains(100));
+	}
+
+	/** Test of the hashCode() method */
+	@Test
+	public void hashCodeTest () {
+		/*
+		 * Testing if the hash code of an ordered long array is in fact the same provided as the hashCode() method from Object.
+		 */
+		Object longArrayUnordered = new LongArray(false, 16);
+		((LongArray)longArrayUnordered).add(2, 1, -6, 2);
+		((LongArray)longArrayUnordered).add(5, 50, 34, 42);
+		Assert.assertEquals(longArrayUnordered.hashCode(), ((LongArray)longArrayUnordered).hashCode());
+
+		LongArray longArrayOrdered = new LongArray();
+		longArrayOrdered.add(10, 20, 40, 50);
+		longArrayOrdered.add(100, 200, 400, 500);
+
+		/*
+		 * Expected hash calculation: 1. Start with h = 1 2. For each value in the array, compute (item ^ (item >>> 32)) and add it
+		 * to h as: h = h * 31 + (int)(item ^ (item >>> 32)) Detailed steps: - Iteration 1 (item = 10): h = 1 * 31 + 10 = 41 -
+		 * Iteration 2 (item = 20): h = 41 * 31 + 20 = 1 291 - Iteration 3 (item = 40): h = 1 291 * 31 + 40 = 40 061 - Iteration 4
+		 * (item = 50): h = 40 061 * 31 + 50 = 1 241 941 - Iteration 5 (item = 100): h = 1 241 941 * 31 + 100 = 38 500 271 -
+		 * Iteration 6 (item = 200): h = 38 500 271 * 31 + 200 = 1 193 508 601 - The next operation will overflow because h is an
+		 * int and can't be calculated if >= 2 147 483 647 (because negative numbers are defined by the first bit - The result of 1
+		 * 193 508 601 * 31 + 400 is 36 998 767 031 (1000 1001 1101 0100 1100 0110 0001 1011 0111) which is, truncated to 32 bits,
+		 * -1 655 938 633 (1001 1101 0100 1100 0110 0001 1011 0111) - Iteration 7 (item = 400): h = - 1 655 938 633 - Same here,
+		 * without redetailing the operation, we get, truncated to 32 bits, 205 510 429 - Iteration 8 (item = 500): h = 205 510 429
```

Comment:
I understand Spotless probably did this, but I find it very difficult to read as the `-` used for bullet points ends up looking like mathematical subtraction. This looks like a complete sentence on first read: `Start with h = 1 2.` You may wish to consider using `<ul>` and `<li>` to create a formatted list. If this was generated with AI and you haven't gone through it yourself, I urge you to just throw it away.

Reply:
Also, for one thing testing hash codes this way doesn't actually provide any validation of any properties of a hashing function... for another thing, hashCode() impls in Java are usually quite simple, and that's fine for their typical usage in hash tables. The main requirement for a hash function to use in a hash table is just for the function to be fast to call (as in, a small bytecode size); most String keys aren't novel-length texts, and a LongArray used as a key in a HashMap probably won't have 1000+ items, so the actual performance of the function per item doesn't matter as much as if it can be inlined early, and the quality can be actually quite bad and still work fine in a hash table.

I think the whole hashCodeTest() should be removed.